V11 v2

cmschuetz opened the pull request:
#### What's this do?

Pulls the latest upstream changes from create-react-app (v2.1)

#### Why are we doing this? (w/ JIRA link if applicable)

@trunkclub/build hasn't been updated in awhile and create-react-app v2.1 provides some nice new features

#### Who should review this PR?

@trunkclub/ui

#### Anything the reviewer(s) should know to review effectively?

Yes, please see https://github.com/trunkclub/tcweb-build/blob/1c0ff3c2afc2c6725e86f7ad4c9c16355c29703f/docs/UPGRADING-v11.md for a list of breaking changes.  If you disagree with any, please reach out.

#### How should this be tested?

An alpha release (@trunkclub/build@11.0.10) is currently used in production for `pos_app`.  I'd like a few other teams to start using these versions as well to see if I missed anything.

#### Dependencies for merging? Releasing to production?

Nope

selbekk committed:
Move using global variables to its own file

selbekk committed:
Move adding bootstrap to its own file

selbekk committed:
Move adding flow to its own file

selbekk committed:
Move adding relay to its own file

selbekk committed:
Move adding a router to its own file

selbekk committed:
Move adding custom environment variables to its own file

selbekk committed:
Move can i use decorators to its own file

selbekk committed:
Move fetching data with ajax requests to its own file

selbekk committed:
Move integrating with an API backend to its own file

selbekk committed:
Move using https in development to its own file

selbekk committed:
Move generating dynamic meta tags on the server to its own file

selbekk committed:
Move pre-rendering static html files to its own file

selbekk committed:
Move injecting data from the server into the page into its own file

selbekk committed:
Move making a progressive web app into its own file

selbekk committed:
Move advanced configuration into its own file

selbekk committed:
Move alternatives to ejecting to its own file

selbekk committed:
Remove the user guide file

selbekk committed:
Create a documentation intro page

selbekk committed:
No markdown in the headings

selbekk committed:
Update all internal links

selbekk committed:
Bumping all headings one level because we can now

selbekk committed:
Remove temporary readme file

selbekk committed:
Merge pull request #5341 from selbekk/add-callouts-and-logo

Add callouts and logo

selbekk committed:
Merge pull request #5344 from selbekk/clean-up-getting-started-page

Add some headings to the getting started section

selbekk committed:
Merge pull request #5347 from selbekk/port-user-guide-over

Port user guide over

selbekk committed:
Add Inc to copyright notice

selbekk committed:
Add underlines for post links

0xdeafcafe committed:
Add environment variable to optionaly disable inlining of chunks (#5354)

* Add environment variable to optionaly disable inlining of chunks

* Changed env var to `process.env.INLINE_RUNTIME_CHUNK `

* Added env var to Advanced Configuration in docs

* Drop implied behavior

* Tweak working to match other descriptions

selbekk committed:
Create a new Styles and Assets sidebar category

selbekk committed:
Create two new categories for the sidebar

selbekk committed:
Combine the supported browsers and language features pages

selbekk committed:
Combine a few articles on setting up your editor

selbekk committed:
Add a note about why you _should_ set up your editor

selbekk committed:
Shorten label

selbekk committed:
Shorten the sidebar labels

selbekk committed:
Combine pages about head tags

selbekk committed:
Combine even more pages

selbekk committed:
Remove API reference from footer

Timer committed:
Fix checking against non-semver compliant Yarn versions (#5362)

Commit:
Title Case All Headings

Commit:
Make 'publishing components to npm' a sub-part of the deployment doc

Commit:
Move testing above back-end integration

Commit:
Make 'sending feedback' a part of the introduction

Commit:
Update i18n file

jonscottclark committed:
Update path to serviceWorker.js in docs (#5371)

Timer committed:
Clean up the behavior tests (#5365)

* Speed up installs with pnp

* Move to a better relative path test

* Continue work on new test organization

* Move mjs test to new enhanced tests

* Move over last legacy test

* Update behavior e2e script

* Add first iteration of instructions to test readme

* Add some more bad instructions

* Split test command into multiple lines

* Use two workers (do not run in band)

* Bump install timeout

* No global pollution

* Fix PnP support via standard run mechanism

* Prevent circular serve command

* Silent yarn execs

* Fix common commands by using a shell

* Run with npx instead of Yarn

* Remove unused scripts

* Wait for localhost and fix script booting with PnP

* Don't pnp locally because links will cause module resolution to whine

* lint-staged is annoying

* Strip Yarn out of execa runs

* Set default license if none specified

* Don't impose worker limit

* Disable pnp for webpack messages

* Add missing dependency

* Disable timeout and rely on Jest

* Only use 2 jest workers

* Add missing dependency

selbekk committed:
Merge pull request #5357 from selbekk/potpourri

Potpourri

Commit:
Make the navigation easier to read on small devices

Commit:
Add section about how to get started

Commit:
Add section about updating

Commit:
Add a few more features to the bottom

Commit:
Fix indentation mess

Commit:
Add missing period

Commit:
Remove punctuation for consistency

amyrlam committed:
Make `Getting Started` consistent

amyrlam committed:
Order docs in footer

Rename to match page "About The Docs"

These seem redundant, consider just "Getting Started"? Add more links?

amyrlam committed:
Edits to Getting Started 

- Copy edits for consistency
- Remove <br>'s for readability

Should all headers be Title Case? I see `Creating an App` but `Installing A Dependency`. I tend to use https://titlecaseconverter.com/, thoughts?

selbekk committed:
Merge pull request #5400 from amyrlam/amy/tweaks

Tweaks

amyrlam committed:
Merge pull request #5398 from selbekk/front-page-content

Add some front page content

Commit:
Fix eslint config docs. (#5416)

Update peer dependency versions in README.

neo committed:
Add the new SVGs feature from the template README to the root README (#5374)

jeffposnick committed:
Updates to clarify the "waiting" SW behavior (#5410)

vikr01 committed:
use the correct dependency for babel-plugin-dynamic-import-node (#4984)

* use the correct dependency for babel-plugin-dynamic-import-node

* Update package.json

gaearon committed:
Fix click-to-open on Windows (#5431)

* Fix click-to-open on Windows

* Oops

gavinbarron committed:
Correct instructions for HTTPS and PowerShell (#5394)

Current instructions for running using HTTPS in a PowerShell prompt are incorrect and do not work on Windows 10 v1809

rlueder committed:
Added note on update to .flowconfig for .scss imports (#5321)

* Added note on update to .flowconfig for .scss imports

Added a note aimed at Flow users on how to get Flow to recognize the .scss imports.

* Mentioned .sass extension in note, added backticks

gaearon committed:
Downgrade lerna-changelog because it has a bug

gaearon committed:
Update cached lockfile

gaearon committed:
Changelog for 2.0.5

gaearon committed:
Publish

 - babel-preset-react-app@5.0.4
 - confusing-browser-globals@1.0.4
 - create-react-app@2.0.4
 - eslint-config-react-app@3.0.4
 - react-dev-utils@6.0.5
 - react-error-overlay@5.0.5
 - react-scripts@2.0.5

ranyitz committed:
Remove dot from the end of the link (#5449)

thompk2 committed:
Fix typo

explictly seems like a typo for explicitly

Commit:
Merge pull request #5475 from thompk2/patch-1

Fix typo

sophiebits committed:
New copy, by me

cmschuetz committed:
feat: Add tc-specific logic

- Adds tc env plugin
- Sets TC_ENV variables correctly
- Modifies webpack dev/prod to import env plugin

cmschuetz committed:
chore: Name package and bump version

cmschuetz committed:
chore: Adds trunkclubVersionsPlugin

- Actually adds trunkclubVersionsPlugin
- Bumps version

gaearon committed:
Merge branch 'docusaurus'

gaearon committed:
Port documentation updates

gaearon committed:
doc tweaks

gaearon committed:
fix url

gaearon committed:
Point README to docs

gaearon committed:
Update README.md

gaearon committed:
Update README.md

gaearon committed:
Update README.md

gaearon committed:
Update README.md

jessepeterman committed:
Fixed typo in getting-started (#5493)

changed amy to any

cmschuetz committed:
fix: Re-add progress bar & modify script name

cmschuetz committed:
chore: Bump version

cmschuetz committed:
docs: v11 Upgrade Guide

- Adds v11 upgrade guide
- Bumps version
- Re-adds missing docs
- Re-adds missing custom scripts

cmschuetz committed:
Merge branch 'master' of github.com:facebook/create-react-app into v11-v2

cmschuetz committed:
chore: Bump version

Commit:
Fix typo (#5494)

cmschuetz committed:
Merge branch 'master' into v11-v2

cmschuetz committed:
fix: Missing pieces

- Re-adds .circleci/config.yml
- Updates config.yml
- Sets env in lint script
- Re-adds publish.js
- Bumps version

Timer committed:
Remove --dev flag

brunolemos committed:
TypeScript syntax support (#4837)

yuyokk committed:
Fix title on Safari (#5499)

iansu committed:
Update doc links in template README (#5512)

gaearon committed:
Update required version in README

amyrlam committed:
Add "edit" feature to Docusaurus pages (#5492)

Timer committed:
Always type check TypeScript when being used (#5515)

* Always type check TypeScript when being used

* Use alternate version

Timer committed:
Check for TypeScript install in preflight (#5516)

* Check for TypeScript install in preflight

* Remove unused import

Timer committed:
Validate tsconfig when using TypeScript (#5524)

* Sanity check TypeScript config

* Check more options

* Set all defaults and suggestions

* Update docs

* Update doc notes

* Automatically copy react app declared types to project on start

* Remove note about loaders.d.ts

Timer committed:
Automatically create a `tsconfig.json` when entrypoint is TypeScript

Timer committed:
Hoist compiler options

Timer committed:
Print friendly message on first time use

Timer committed:
Fix TypeScript file detection

Timer committed:
Properly exclude test files from typechecking

Timer committed:
Add an advanced typescript test case

jeffposnick committed:
Update Workbox dependency (#5527)

Timer committed:
Add TypeScript error formatting (#5529)

Timer committed:
Update adding-typescript.md

brunolemos committed:
Enable TypeScript json module resolver (#5531)

* Enable TypeScript json module resolver

* Update verifyTypeScriptSetup.js

brunolemos committed:
Use TypeScript parser to read tsconfig.json (#5532)

brunolemos committed:
Simplified TypeScript steps (#5533)

ianschmitz committed:
Respect tsconfig.json extends when validating config (#5537)

* Use TS to resolve tsconfig extends

* Prevent modifications to original tsconfig

* Print friendly error

NShahri committed:
Avoid pushing .pnp folder to git (#5469)

* Add .pnp folder ro gitignore file

Avoid pushing .pnp folder to git

* Add .pnp.js to ignore list

Timer committed:
Remove service worker types from the main install

Timer committed:
Automatically setup TypeScript when detected (#5549)

* Don't resolve TS files if it's not detected in the project

* Automatically create tsconfig.json file for user

* Remove always false check

* Add missing file

* Don't filter paths too early

amyrlam committed:
Add Algolia search bar to Docusaurus (#5551)

Timer committed:
Simplify TypeScript setup instructions

Timer committed:
Adjust notes

brunolemos committed:
Change import syntax from typescript declaration (#5559)

brunolemos committed:
Add typings for process.env (#5557)

Timer committed:
Add TypeScript app creation (#5550)

* Add TypeScript app creation

* Actually specify new extension

* Check for ts and tsx

* Fix types in default service worker file

* Mirror changes in JS version of SW

* Separate templates

* Use separate template directory

* Remove unused function

* Add a tsconfig file to the template

* Test install of typescript

* Add e2e for TypeScript

* Check for index.tsx to make sure we're not in the JS template

* Ensure TypeScript doesn't leak into normal installs

* Ignore some files

* Print version of CRA

* Fix script

* Add new template

* Move test to correct location

* Use `verifyTypeScriptSetup` routine to populate files

* Ensure tsconfig is created

Timer committed:
Don't auto exclude test files

Timer committed:
Enable forceConsistentCasingInFileNames by default

Timer committed:
Fix tests

Timer committed:
Fix tests x2

Timer committed:
Remove publish time waste

Timer committed:
Always publish with npm

Timer committed:
Add instructions for a prerelease

Timer committed:
Prompt for 2fa auth code

Timer committed:
Update lockfile

Timer committed:
Check for CI in publish

brunolemos committed:
Update App.tsx (#5588)

jheijmans committed:
Fixing the code splitting links in the ESLint output (#5586)

brunolemos committed:
Fix process.env types (#5589)

Fix #5576

Timer committed:
Refine how TypeScript env types are handled (#5593)

* Specify types in package

* Do not remove types file on eject

* Stop copying types into generated project

* Reference react and react-dom

* Reference node types

* Install node types as well

* Restore copying

* Add Node to the list of installed types

* Reference Jest types

* Remove jest types from install

* Remove jest from CRA install

* Remove Jest reference and let user do this themselves

* Stop copying types file

* Add types key to package.json

* Add appTypeDeclarations and create when missing

* Rename declarations file

* Add Jest back to install instructions

* Minimize diff

Timer committed:
Turn on certain TypeScript options (#5607)

Timer committed:
Ignore test files from reported type errors (#5608)

* Ignore test files from reported type errors

* Ignore setup proxy & test files

Timer committed:
Remove unsupported TS options (#5609)

Timer committed:
Compile lock file on publish

brunolemos committed:
Ignore json files from TypeScript type checking (#5614)

Fix #5613

ehfeng committed:
Fixing Internal Links (#5552)

* Fixing Internal Links

Internal links, like those found on [Available Scripts](https://facebook.github.io/create-react-app/docs/available-scripts)
use absolute urls and therefore link to https://facebook.github.io/docs/deployment instead of
https://facebook.github.io/create-react-app/docs/deployment.

* changing to markdown links and fixing various broken internal links

Timer committed:
Remove react-scripts type reference on eject (#5611)

* Remove react-scripts type reference on eject

* Check for env file

* Check eject for typescript

* Shuffle appTypeDeclarations

* Append internal types on eject

* Ensure lib is published for types

* Adjust comment

* Don't add a bunch of new lines

* File should exist and not be deleted

* Add debug

* Set file explicitly

* Revert "Set file explicitly"

This reverts commit bcd58a36cbd08a71af50b037d8f1fae6c595fb4e.

* Copy file before destroying ourselves

* Revert "Add debug"

This reverts commit 0068ba81c6d79d99788877c9e1b618acd7412dce.

Timer committed:
Update lockfile

Timer committed:
Add release documentation

Timer committed:
Update lockfile

Timer committed:
Publish

 - babel-preset-react-app@6.0.0
 - confusing-browser-globals@1.0.5
 - create-react-app@2.1.0
 - eslint-config-react-app@3.0.5
 - react-dev-utils@6.1.0
 - react-error-overlay@5.0.6
 - react-scripts@2.1.0

Timer committed:
Add note about react-scripts version

Timer committed:
Update docs url

Vinnl committed:
Add link to cra-ts migration guide (#5629)

For users coming from create-react-app-typescript, the added link provides instructions on how to port their app to Create React App.

iansu committed:
Update minimum node version to 8.10 in README (#5635)

cmschuetz committed:
Merge branch 'master' of github.com:facebook/create-react-app into v11-v2

cmschuetz committed:
chore: Bump version

xiaoxiangmoe committed:
fix: Duplicate string index signature in ProcessEnv (#5621)

eddedd88 committed:
Make serviceWorker config argument optional in template (#5651)

NShahri committed:
add tests for named-asset-imports plugin (#5575)

* add tests for named-asset-imports plugin

* add ticketFormats store/saga

* change import to require on test file to avoid using babel when running tests

* remove all babel dependencies as not needed more for running tests

* update tests and rename file to index.test.js

* remove npmignore as it is whitelisted in package.json

* add babel-plugin-named-asset-import tests to e2e

jakeboone02 committed:
Update making-a-progressive-web-app.md (#5658)

Timer committed:
Add support for decorators (#5659)

* Turn on decorators

* Add decorator test

hipstersmoothie committed:
Expose `reportRuntimeError` (#4709)

* factor out crashWithFrames and expose reportRuntimeError

* address code review and move error call to handleRuntimeError

Timer committed:
Add changelog for 2.1.1

Timer committed:
Update cached lock file

Timer committed:
Publish

 - babel-plugin-named-asset-import@0.2.3
 - babel-preset-react-app@6.1.0
 - create-react-app@2.1.1
 - react-dev-utils@6.1.1
 - react-error-overlay@5.1.0
 - react-scripts@2.1.1

cmschuetz committed:
Merge branch 'master' of github.com:facebook/create-react-app into v11-v2

cmschuetz committed:
chore: Bump version

cmschuetz committed:
chore: Update upgrade guide

iansu committed:
Add bot config (#4483)

* Add stale config

* Update stale config

* Add lock config

* Update lock.yml

tabrindle committed:
update envinfo to 5.11.1 (#5685)

denofevil committed:
ignore intellij module files when generating an app (#4605)

Timer committed:
Make stale bot configuration more aggressive

Timer committed:
Lock issues more aggressively

Timer committed:
Add permissive TS lib defaults (#5694)

ianschmitz committed:
Fix tsconfig.json lib suggested value (#5701)

makovkastar committed:
Fix broken documentation link (#5670)

Fix public folder documentation link

m4jing committed:
Fixed link to manifest.json file (#5704)

githubsaturn committed:
Updated the link to firebase hosting (#5710)

imjoshdean-tc commented on packages/react-scripts/scripts/build.js:
```diff
@@ -1,17 +1,17 @@
 'use strict';
 
-// Do this as the first thing so that any code reading it knows the right env.
+/** TC CUSTOM */
```
Two quick suggestions.

**1.) Add _why_ we do this.** I love that we are marking sections that are clearly identifiable as Trunk Club custom code, but having a little insight as to why we are doing this will be helpful for others coming and looking at this code.

**2.) Call this a deprecation and suggest an alternative means to achieve this.** If one of the goals at this point is to make it possible for our projects to use vanilla CRA, we should do everything we can to get out of the way from them doing that. In this case, perhaps it's chalking out that we shouldn't rely on `TC_ENV` and that developers should use `REACT_APP_ENV` if they need to identify differences between staging and production.

imjoshdean-tc commented on packages/react-scripts/utils/trunkclubVersionsPlugin.js:
```diff
@@ -1,65 +1,65 @@
-'use strict'
+'use strict';
```
How important is this file? It seems like we could potentially get rid of it and not have lost much if anything.

imjoshdean-tc replied:
There's one sourced use the output of this - https://github.com/trunkclub/stylist_app/blob/master/src/lib/pollForAppUpdates.js

I wonder if sales could be convinced to either not have this feature or opt to eject long-term.

imjoshdean-tc replied:
I spoke with them and I think what might be the wisest move would be to include this in v11, have them look into an alternative approach to their use for it, and if they need to use it and want to move to vanilla CRA, they could eject and use this then.

imjoshdean-tc commented on packages/react-scripts/config/webpack.config.prod.js:
```diff
@@ -1,38 +1,52 @@
+// @remove-on-eject-end
 
-const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+/** TC CUSTOM */
```
Same comment as the one for the previous TC CUSTOM section.

imjoshdean-tc commented on packages/react-scripts/config/webpack.config.dev.js:
```diff
@@ -1,46 +1,102 @@
+
+/** TC CUSTOM */
+const TrunkClubVersionsPlugin = require('../utils/trunkclubVersionsPlugin');
+const ProgressBarPlugin = require('progress-bar-webpack-plugin');
```
Generally agreed upon that we can get rid of this. Let's give it the 🔪

BPScott committed:
Version bump postcss-preset-env to latest (#5721)

Commit:
Fix typo (#5727)

cmschuetz committed:
Merge branch 'master' of github.com:facebook/create-react-app into v11-v2

cmschuetz committed:
docs: Adds documentation for custom logic

- Adds documentation giving context around custom logic and suggestions
for moving forward without including them in this fork

cmschuetz committed:
chore: Bumps version

imjoshdean-tc reviewed (approved):
Hell yeah

Timer committed:
Update stale.yml

Timer committed:
Tweak bot settings

Timer committed:
Merge webpack configuration (#5722)

* Rename production configuration

* Upgrade lint staged

* Merge mode, bail, devtool, and entry

* Merge output settings

* Remove old comment

* Merge some trivial differences

* Disable minimize in development mode

* Only minify html and generate SW in production

* Adjust comment

* Toggle development plugins

* Add missing imports

* Scope settings

* Delete development config

* Use new config file

* Remove unnecessary config import

dunglas committed:
Suggest Encore when not building a SPA with Symfony (#5730)

* Suggest Encore when not building a SPA with Symfony

* Update README.md

open-source-explorer committed:
Add PR welcoming badge  (#5759)

* add pr badge :tada:

* add rounded badge :boom:

* Update README.md

andys8 committed:
docs: Change Storybook install documentation (#5779)

gshilin committed:
Don't polyfill fetch for Node -- additional files (#5789)

sagirk committed:
docs: Simplify installing Storybook with npx (#5788)

Reduce two-step install to just one step (per the latest storybook docs)

Change (i) `npm install -g @storybook/cli` and (ii) `sb init` to `npx -p @storybook/cli sb init`

Refs: https://storybook.js.org/basics/quick-start-guide/

kulek1 committed:
Rename 'getting started' link to 'docs' (#5806)

ajwann committed:
extra polyfills must be included manually (#5814)

* extra polyfills must be included manually, fixes #5795

* make note italicized

pavinthan committed:
React native repository updated in README.md (#5849)

Create React Native App (https://github.com/react-community/create-react-native-app ) has been archived by the owner. It is now read-only.

NShahri committed:
Make named-asset-import plugin work with export-as syntax (#5573)

* update named-asset-import babel plugin

* add tests to named-asset-import plugin

* add more plugin test

* extract generateNewSource method, renaming variables

* extract replaceNotVisitedSpecifiers for export and import visitors

* remove visited list update from importDeclaration

* renaming methods and removing return directly instead of saving

Pajn committed:
Add allowESModules option to babel-preset-react-app (#5487)

* Add allowESModules option to babel-preset-react-app

* changes after feedback

* Apply suggestions from code review

Co-Authored-By: Pajn <rasmus@eneman.eu>

SimenB committed:
fix: add `sideEffects: false` to react-error-overlay (#5451)

This allows us to leave the import in the code, and webpack will still tree shake it out

ianschmitz committed:
Fix TypeScript decorator support (#5783)

* Fix TypeScript decorator support

* Update babel flow override

* WIP

Timer committed:
Fix annotated var test

Timer committed:
Upgrade babel deps

Timer committed:
Fix propertyDecorator test

Commit:
Always test with the latest stable Node version on Travis (#5546)

SimenB committed:
fix: make typescriptformatter support 0.5 of fork checker (#5879)

wsmoak committed:
Fix link to page about running tests (#5883)

nikhilknoldus committed:
Some Grammar fixes (#5858)

* Grammar fixes

1- For `Here's a `: The singular verb form ’s does not seem to agree with the plural subject a few common cases where you might want to try something else. Consider changing the verb form.

2- Added `A` before Reasonably : The noun phrase Reasonably good configuration seems to be missing a determiner before it. Consider adding an article. An article (a, an, or the) is a type of determiner. Possessive adjectives (my, his, our), possessive nouns (Joe’s, mother’s), and quantifiers (each, every) are also determiners. Single countable nouns usually require a determiner.

* Update README.md

sibiraj-s committed:
Run prettier on HTML files (#5839)

iansu committed:
Remove unnecessary whitespace in template HTML

ianschmitz committed:
Support setupTests.ts (#5698)

heyimalex committed:
Disable copy to clipboard in cra --info (#5905)

* Disable copy to clipboard in cra --info

Fixes an issue where npx create-react-app --info throws an exception on Windows. The exception was caused by envinfo's copy to clipboard functionality; envinfo is distributed pre-bundled, but the dependency it uses for copying to the clipboard makes use of a binary on Windows that can't be found once the bundle is made. Disabling copy to clipboard fixes the issue. Closes #5757.

* Explicitly set clipboard to false

iansu committed:
Add placeholders to template README for bit.ly links. (#5808)

* Add placeholders to template README for bit.ly links.

* Add placeholders to template README for goo.gl links.

wuweiweiwu committed:
Update PWA link in README (#5907)

alaycock committed:
Add note to docs about using Sass and Flow together (#5823)

* Add default values to `file_ext` note

* Update note to include a link to the relevant docs

* Update copy to indicate the default values

* Add missing space

mjackson committed:
Correct some comments (#5927)

And remove some outdated info from 1.x

swyxio committed:
Add netlify.toml to prepare for deploy to netlify #5807 (#5930)

* commit toml

* full build path

* wow the path is long

* redirect

iansu committed:
Add pre-eject message about new features in v2 (#5954)

* Add pre-eject message about new features in v2

* Make message bold

Timer committed:
Suggest a different default for speed reasons (#5959)

Advanced users can opt into this behavior, but it's generally outside the users' control to fix and comes with performance concerns

jayantbh committed:
Add SASS_PATH instructions to Sass docs (#5917)

`node-sass` supports the SASS_PATH variable that we can use. This PR adds that to the documentation.

phacks committed:
Add removeItem to localStorage mock in docs (#5919)

stephen-algeo committed:
Improve error messaging in verifyPackageTree.js (#5974)

iansu committed:
Use https for linked images in docs to fix mixed content warnings (#5985)

iansu committed:
Update links to docs in all package README files (#5912)

* Update links to docs in all package README files

* Fix formatting of issue template

* Fix formatting of issue template

ManoelLobo committed:
Added extension to .eslintrc (#5988)

ManoelLobo committed:
Update eslint instructions in docs (#5990)

ianschmitz committed:
Add production build section to docs (#5900)

iansu committed:
Make links to docs consistent in README (#6000)

mbelsky committed:
Add docs for creating new TypeScript project (#6015)

dugagjin committed:
add webp support for typescript (#5978)

* add webp support for typescript

* moved webp declaration below png

souzasmatheus committed:
Fix Markdown comment in proxying docs (#6009)

cmschuetz committed:
Merge branch 'master' of github.com:facebook/create-react-app into v11-v2

cmschuetz committed:
Chore: Bumps version

Timer committed:
Tweak contributing doc

Timer committed:
Prepare 2.1.2 release

Timer committed:
Publish

 - babel-plugin-named-asset-import@0.3.0
 - babel-preset-react-app@7.0.0
 - create-react-app@2.1.2
 - eslint-config-react-app@3.0.6
 - react-app-polyfill@0.2.0
 - react-dev-utils@7.0.0
 - react-error-overlay@5.1.1
 - react-scripts@2.1.2

Timer committed:
Add note to changelog

Commit:
Fix control comment fo CSS Grid prefixing (#6061)

"on" value of "autoprefixer grid" is kinda deprecated. It's better to use "autoplace" or "no-autoplace" instead.
https://github.com/postcss/autoprefixer#options

"autoplace" enables PostCss' autoplace polyfill for ie and makes prefixes, "no-autoplace" only makes prefixes
https://github.com/postcss/autoprefixer#does-autoprefixer-polyfill-grid-layout-for-ie

LukasWerfel committed:
Update eslint instructions in docs (#6084)

Add a note that it will work in IDE with 2.0.3 and later

jquense committed:
Generalize the Bootstrap documentation (#5631)

* Generalize the adding bootstrap documentation

* Update adding-bootstrap.md

* Update adding-bootstrap.md

* Update adding-bootstrap.md

iansu committed:
Update lock bot config

prashant-andani committed:
Typo fixes (#6104)

<!--
Thank you for sending the PR!

If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots!

Happy contributing!
-->

shawtung committed:
Fix comment typo (#6036)

raiskila committed:
Replace deprecated VSCode launch.json variable (#6110)

Replace deprecated ${workspaceRoot} variable with ${workspaceFolder} in VSCode launch.json. See https://code.visualstudio.com/docs/editor/variables-reference#_why-isnt-workspaceroot-documented

Friss committed:
Update webpack-dev-server 3.1.9 -> 3.1.14 (#6064)

* Update webpack-dev-server 3.1.9 -> 3.1.14

gaearon committed:
Update adding-typescript.md

Commit:
Fix typo in CSS modules docs (#6067)

ianschmitz committed:
Prepare 2.1.3 release

ianschmitz committed:
Publish

 - create-react-app@2.1.3
 - react-dev-utils@7.0.1
 - react-error-overlay@5.1.2
 - react-scripts@2.1.3

ianschmitz committed:
Update CHANGELOG.md

cmschuetz committed:
Merge commit '73e3d0ebf1f2834e1c8c41d3a25ae5e0e99e6f14' into v11-v2

cmschuetz committed:
chore: Release version
